### PR TITLE
Add dropout2d

### PR DESCRIPTION
**Pull request by deschena**
## Proposed changes

I broke the PR https://github.com/ml-explore/mlx/pull/222 with `cherry-pick` operations. I am sorry for the inconvenience. Here are the changes again..

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: python/mlx/nn/layers/dropout.py [hunk 1]
```diff
@@ -32,4 +32,61 @@ def __call__(self, x):
...
+class Dropout2d(Module):
+    """Apply 2D channel-wise dropout during training.
+
+    Randomly zero out entire channels independently with probability :math:`p`.
+    This layer expects the channels to be last, i.e. the input shape should be
+    ``NWHC`` or ``WHC`` where:
+        - ``N`` is the batch dimension
+        - ``H`` is the input image height
+        - ``W`` is the input image width
+        - ``C`` is the number of input channels
```
Comment (awni) [inline python/mlx/nn/layers/dropout.py:47]: 👍

File: python/mlx/nn/layers/dropout.py [hunk 2]
```diff
@@ -32,4 +32,61 @@ def __call__(self, x):
...
+
+    The remaining channels are scaled by :math:`\frac{1}{1-p}` to
+    maintain the expected value of each element. Unlike traditional dropout,
+    which zeros individual entries, this layer zeros entire channels. This is
+    beneficial for earlyconvolution layers where adjacent pixels are
```
Comment (awni) [inline python/mlx/nn/layers/dropout.py:52]: typo `early convolution`

**Review by awni — APPROVED** [review]
Looks great!! could you fix the typo, then i'll merge it?

**Comment by deschena** [comment]
Thank you very much for reviewing again, I have now fixed the typo

**Pull request merged**